src/index.ts

```diff
@@ -93,142 +91,136 @@ function createExtensions(document, result, requestID) {
-      userAgent,
-      appToken,
-    }
-
-    const validationRules = [
-      principalFieldDirectiveValidation,
-
-      // require Authorization header for introspection (in production if configured)
-      ...(INTROSPECT_TOKEN &&
-      req.headers["authorization"] !== `Bearer ${INTROSPECT_TOKEN}`
-        ? [NoSchemaIntrospectionCustomRule]
-        : []),
-    ]
-    if (QUERY_DEPTH_LIMIT) validationRules.push(depthLimit(QUERY_DEPTH_LIMIT))
-
-    return {
-      schema: appSchema,
-      graphiql: !PRODUCTION_ENV,
-      context,
-      rootValue: {},
-      customFormatErrorFn: graphqlErrorHandler(enableSentry, {
-        req,
-        // Why the checking on params? Do we reach this code if params is falsy?
-        variables: params && params.variables,
-        query: (params && params.query)!,
-      }),
-      validationRules,
-      extensions: ({ document, result }) =>
-        createExtensions(document, result, requestID),
-    }
+  const loaders = createLoaders(accessToken, userID, {
+    requestIDs,
+    userAgent,
+    appToken,
   })
 
-  app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
+  const exchangeSchema = executableExchangeSchema(legacyTransformsForExchange)
 
-  if (ENABLE_GRAPHQL_UPLOAD) {
-    app.use(
-      graphqlUploadExpress({
-        maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
-        maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
-      })
-    )
+  const context: ResolverContext = {
+    accessToken,
+    userID,
+    defaultTimezone,
+    ...loaders,
+    // For stitching purposes
+    exchangeSchema,
+    requestIDs,
+    userAgent,
+    appToken,
   }
 
-  app.use(graphqlServer)
-
-  if (enableSentry) {
-    app.use(Sentry.Handlers.errorHandler())
+  const validationRules = [
+    principalFieldDirectiveValidation,
+
+    // require Authorization header for introspection (in production if configured)
+    ...(INTROSPECT_TOKEN &&
+    req.headers["authorization"] !== `Bearer ${INTROSPECT_TOKEN}`
+      ? [NoSchemaIntrospectionCustomRule]
+      : []),
+  ]
+  if (QUERY_DEPTH_LIMIT) validationRules.push(depthLimit(QUERY_DEPTH_LIMIT))
+
+  return {
+    schema,
+    graphiql: !PRODUCTION_ENV,
+    context,
+    rootValue: {},
+    customFormatErrorFn: graphqlErrorHandler(enableSentry, {
+      req,
+      // Why the checking on params? Do we reach this code if params is falsy?
+      variables: params && params.variables,
+      query: (params && params.query)!,
+    }),
+    validationRules,
+    extensions: ({ document, result }) =>
+      createExtensions(document, result, requestID),
   }
+})
+
+app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
 
-  return app
+if (ENABLE_GRAPHQL_UPLOAD) {
+  app.use(
+    graphqlUploadExpress({
+      maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+      maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
+    })
+  )
 }
 
-const app = express()
+app.use("/v2", (req, res, next) => {
+  if (req.url !== "/") return next()
```

Comment:
Kind of weird but paths like "/v2/blah/anythin" get matched and respond to GraphQL queries (?) , so this ensures those 404, and only "/v2" matches (with an optional trailing slash at the end like "/v2/")

Reply:
It's surprising to me that this special handling is necessary to avoid matching by prefix. (Is this because of `use` vs. just `get` or `post`?)

Reply:
Yea I wasn't 100% sure what was up with that behavior, I'm assuming you are right b/c `get` would naturally need to support query params.

Dropping support for GET requests feels like its own cleanup perhaps.